packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -39,6 +39,19 @@ export const useGridSelection = (apiRef: GridApiRef, props: GridComponentProps):
     return [props.selectionModel];
   }, [props.selectionModel]);
 
+  useGridStateInit(apiRef, (state) => ({ ...state, selection: propSelectionModel ?? [] }));
+
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const rowsLookup = useGridSelector(apiRef, gridRowsLookupSelector);
+
+  const isStateControlled = useGridRegisterControlState(apiRef, {
```

Comment:
This new `useGridRegisterControlState` hook has two concerns: registering the state and checking if the prop is controlled. I would call `updateControlState` as before and check the prop to see if it is controlled. Its second purpose is only used once and here.

Reply:
What do you mean by "call updateControlState as before"

I can remove the 2nd purpose to simplify the understanding of this hook.
The important point of this hook is that it registers synchronously the controlled state rather than in an effect.

Reply:
I was saying that we could remove `useGridRegisterControlState` and call `apiRef.current.updateControlState` directly, but in a synchronous way. The new hook seems to be only a wrapper.

Reply:
I thought that it would eventually have more logic inside it, but the PR is ready and there is none so I suppose we can indeed remove it :+1: